### PR TITLE
[test] Fix disk pressure, run in parallel

### PR DESCRIPTION
**Pull request by WVerlaek**
## Description

* Run e2e tests in parallel
* Add 2 local SSDs to the preview VM if `with-large-vm` is enabled, which together with [this change](https://github.com/gitpod-io/gitpod-packer-gcp-image/pull/240) removes node disk pressure issues, which were causing the tests to fail
* Some other fixes + improvements, see self review

## Related Issue(s)

Relates to WKS-228 WKS-153 WKS-227

## How to test

Create preview env with large vm:
```
TF_VAR_with_large_vm=true leeway run dev:preview
```

Run tests:
```
cd test
./run.sh -s workspace
```

See also two successful GHA runs:
* https://github.com/gitpod-io/gitpod/actions/runs/5442586097
* https://github.com/gitpod-io/gitpod/actions/runs/5443292092

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: .github/workflows/workspace-integration-tests.yml [hunk 1]
```diff
@@ -177,44 +177,31 @@ jobs:
...
-                      # running tests in parallel saves time, but is flakey.
-                      go test -p 1 --parallel 1 -v ./... "${args[@]}" -run '.*[^.SerialOnly]$' 2>&1 | go-junit-report -subtest-mode=exclude-parents -set-exit-code -out "TEST-${TEST_NAME}-PARALLEL.xml" -iocopy
-                      RC=${PIPESTATUS[0]}
-                      set -e
-
-                      if [ "${RC}" -ne "0" ]; then
-                        FAILURE_COUNT=$((FAILURE_COUNT+1))
-                      fi
-                  done
+                  WORKSPACE_TEST_LIST="$CONTENT_SERVICE_TESTS $IMAGE_BUILDER_TESTS $WS_DAEMON_TESTS $WS_MANAGER_TESTS $WORKSPACE_TESTS"
```
Comment (WVerlaek) [inline .github/workflows/workspace-integration-tests.yml:180]: run these packages in parallel

File: .github/workflows/workspace-integration-tests.yml [hunk 2]
```diff
@@ -177,44 +177,31 @@ jobs:
...
+                  if [ "${RC}" -ne "0" ]; then
+                    FAILURE_COUNT=$((FAILURE_COUNT+1))
+                  fi
 
                   exit $FAILURE_COUNT
             - uses: actions/upload-artifact@v3
               with:
                   name: test-results
-                  paths: "test/tests/**/TEST-*.xml"
+                  path: "test/**/TEST-*.xml"
```
Comment (WVerlaek) [inline .github/workflows/workspace-integration-tests.yml:198]: fix: `paths`->`path`

File: dev/preview/infrastructure/modules/gce/variables.tf [hunk 3]
```diff
@@ -44,7 +44,7 @@ variable "harvester_ingress_ip" {
 variable "vm_image" {
   type        = string
   description = "The VM image"
-  default     = "gitpod-k3s-202306121532"
+  default     = "gitpod-k3s-202306301112"
```
Comment (WVerlaek) [inline dev/preview/infrastructure/modules/gce/variables.tf:47]: See https://github.com/gitpod-io/gitpod-packer-gcp-image/pull/240

File: dev/preview/infrastructure/modules/gce/vm.tf [hunk 4]
```diff
@@ -70,8 +81,9 @@ data "kubernetes_secret" "harvester-k3s-dockerhub-pull-account" {
 }
 
 locals {
+  vm_name = "preview-${var.preview_name}"
```
Comment (WVerlaek) [inline dev/preview/infrastructure/modules/gce/vm.tf:84]: prefixing the VM name with `preview-` such that the packer image sets up the preview LVM volumes, which conditionally runs based off the hostname (see https://github.com/gitpod-io/gitpod-packer-gcp-image/pull/240/files#diff-167633e1acfa7e4989f5ba35a6f0847b825b2da186c0a0dd9b1f5a12ff752202R4)

File: test/pkg/integration/setup.go [hunk 5]
```diff
@@ -91,7 +92,7 @@ func Setup(ctx context.Context) (string, string, env.Environment, bool, string,
 	flagset.StringVar(&username, "username", os.Getenv("USER_NAME"), "username to execute the tests with. Chooses one automatically if left blank.")
 	flagset.BoolVar(&enterprise, "enterprise", false, "whether to test enterprise features. requires enterprise lisence installed.")
 	flagset.BoolVar(&gitlab, "gitlab", false, "whether to test gitlab integration.")
-	flagset.BoolVar(&parallel, "parallel", false, "Run test features in parallel")
+	flagset.BoolVar(&parallel, "parallel-features", false, "Run test features in parallel")
```
Comment (WVerlaek) [inline test/pkg/integration/setup.go:95]: `-parallel` overlaps with the equally named built-in go test flag, but that one expects an int value while here we expect a boolean, so renaming this flag such that we can actually set it

File: test/pkg/integration/setup.go [hunk 6]
```diff
@@ -209,39 +210,43 @@ func waitOnGitpodRunning(namespace string, waitTimeout time.Duration) env.Func {
...
 	if err != nil {
-		t.Logf("failed to list pods: %v", err)
+		t.Logf("failed to list pods to log gitpod status: %v", err)
 		return
 	}
 
+	var buf strings.Builder
+	tw := tabwriter.NewWriter(&buf, 0, 0, 1, ' ', 0)
+	_, _ = tw.Write([]byte("Component\tPod\tReady\tStatus\tRestarts\tAge\n"))
+
```
Comment (WVerlaek) [inline test/pkg/integration/setup.go:223]: using a tabwriter makes it easier to read the logs by having all items aligned, see e.g. https://github.com/gitpod-io/gitpod/actions/runs/5423326857/jobs/9861495589#step:8:1646

File: test/pkg/integration/workspace.go [hunk 7]
```diff
@@ -30,7 +30,7 @@ import (
 const (
 	gitpodBuiltinUserID             = "builtin-user-workspace-probe-0000000"
 	perCallTimeout                  = 5 * time.Minute
-	ParallelLunchableWorkspaceLimit = 4
+	ParallelLunchableWorkspaceLimit = 10
```
Comment (WVerlaek) [inline test/pkg/integration/workspace.go:33]: works reliably in a large vm. Ran the workspace test suite 5 times without failure

File: test/tests/components/ws-manager/prebuild_test.go [hunk 8]
```diff
@@ -187,7 +187,7 @@ const (
 // - stop the regular workspace
 // - relaunch the regular workspace
 // - make sure the file foobar.txt exists
-func TestOpenWorkspaceFromPrebuildSerialOnly(t *testing.T) {
```
Comment (WVerlaek) [inline test/tests/components/ws-manager/prebuild_test.go:190]: no more need to run this in serial. Other PRs have already made fixes to this test to make it run reliably

**Review by aledbf — APPROVED** [review]

**Pull request merged**